api/handler.js

```diff
@@ -1,32 +1,36 @@
 'use strict';
 const fetch = require('node-fetch');
-const BASE_URL = 'https://www.york.ac.uk/search/?collection=york-uni-courses&form=course-search&profile=_default&query=';
+const {coursesUrl} = require('./utils/constructFunnelbackUrls');
+const {success, error} = require('./utils/format');
+const ClientError = require('./errors/ClientError');
 
 module.exports.courses = async event => {
+  try {
+    const url = coursesUrl(event.queryStringParameters);
```

Comment:
```suggestion

    if (!event.queryStringParameters || !event.queryStringParameters.search) {
        return error('The search parameter is required.', 400, 'Bad Request', event.path);
    }
    const url = coursesUrl(event.queryStringParameters);
```
By placing the validation of the client's request in the handler we can more easily adjust the validation in future, and maybe drop the complexity from multiple error types in the 'catch' block below. Some interesting discussion on when to throw exceptions vs returning null is [here](https://softwareengineering.stackexchange.com/questions/228287/returning-null-or-a-empty-value-throw-exception).

Reply:
I can see where you're coming from but I don't know if it'd provide any real benefit at this stage - if we're going to work on request validation in the future I suspect we might need to move all this into a separate file anyway?

Reply:
It's to pull the validation and exception throwing out of the URL builder, as right now we've got a function that should be doing pure logic (taking parameters and turning them into a URL) with side effects (validating the user request and throwing an exception). If we do need to move it into a separate file it should also be easier to do from handler since it's more visible.

Reply:
I've created technical debt story CFIN-288 to deal with this issue in the future, following a discussion about meeting the API milestone today.